contrib_bots/lib/CommuteBot/CommuteBot.py

```diff
@@ -0,0 +1,219 @@
+                arrival = params['arrival_time']
+                times = arrival.split(',')
+                times = [int(str(x)) for x in times]
+                a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                b = dt.datetime(times[0],
+                                times[1],
+                                times[2],
+                                times[3],
+                                times[4],
+                                times[5])
+                total_seconds = str(int((b-a).total_seconds()))
+                params.update({'arrival_time': total_seconds})
+
+            # adds API Authentication Key to url request
+            config = cp.ConfigParser()
+            settings = open('lib/settings.ini')
+            config.readfp(settings)
+            apik = config.get('Google.com', 'apik')
+            params.update({'key': apik})
+
+            # sends url api request and converts response into json format
+            r = (requests.get('https://maps.googleapis.com/maps/api/'
+                              + 'distancematrix/json',
+                              params=params))
+            rjson = r.json()
+
+            # determines if user has valid inputs
+            try:
+                not_found = (rjson["rows"][0]["elements"][0]["status"]
+                             == "NOT_FOUND")
+                invalid_request = (rjson["status"] == "INVALID_REQUEST")
+                no_result = (rjson["rows"][0]["elements"][0]["status"]
+                             == "ZERO_RESULTS")
+
+                if no_result:
+                    self.send_info(message,
+                                   "Zero results\n"
+                                   + "If stuck, try '@commute help'.", client)
+                    return None
+
+                elif not_found or invalid_request:
+                    raise IndexError
+            except IndexError:
+                self.send_info(message,
+                               "Invalid input, please see instructions."
+                               + "\nIf stuck, try '@commute help'.", client)
+                return None
+
+            # origin and destination strings
+            begin = 'From: ' + rjson["destination_addresses"][0]
+            end = 'To: ' + rjson["origin_addresses"][0]
+            distance = ('Distance: '
+                        + rjson["rows"][0]["elements"][0]["distance"]["text"])
+            duration = ('Duration: '
+                        + rjson["rows"][0]["elements"][0]["duration"]["text"])
+            output = begin + '\n' + end + '\n' + distance
+
+            # determines if fare information is available
+            try:
+                fare = (
+                        'Fare: '
+                        + rjson["rows"][0]["elements"][0]["fare"]["currency"]
+                        + rjson["rows"][0]["elements"][0]["fare"]["text"])
+                output += '\n' + fare
+            except (KeyError, IndexError):
+                print('')
```

Comment:
Yep. What you are doing is OK too. If you want to ignore the exception you could just have a `pass` statement.